Madness/Parser.swift

```diff
@@ -105,6 +105,23 @@ func containsAt<C1: CollectionType, C2: CollectionType where C1.Generator.Elemen
 	return zip(range, needle).lazy.map { collection[$0] == $1 }.reduce(true) { $0 && $1 }
 }
 
+public func satisfy<C: CollectionType> (pred: C.Generator.Element -> Bool) -> Parser<C, C.Generator.Element>.Function {
+	return { input, index in
+		if input.count > 0 && index != input.endIndex {
```

Comment:
I think this can be reduced to `index < input.endIndex` or at least `index != input.endIndex`.